### ISSUE TITLE
False-positive E1101 (no-member) for `numpy.dtypes.StringDType`

### ISSUE THREAD

**Issue by bersbersbers** [pylint-dev/pylint#9956]
### Bug description

```python
"""Bug."""
import numpy as np
np.array("X", dtype=np.dtypes.StringDType)
```

### Command used

```shell
pylint Bug.py
```

### Pylint output

```python
************* Module bug
bug.py:3:20: E1101: Module 'numpy.dtypes' has no 'StringDType' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.3.0
astroid 3.3.4
Python 3.12.6 (tags/v3.12.6:a4a2d2b, Sep  6 2024, 20:11:23) [MSC v.1940 64 bit (AMD64)]
```

### OS / Environment

Windows 11

### Additional dependencies

```python
numpy==2.1.1
```
